fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/CreditReportWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,218 @@
+                        dataValidationErrors);
+            }
+
+            if (Objects.equals(creditBureauName.get(), CreditBureauConfigurations.THITSAWORKS.toString())) {
+
+                // CreditBureauToken creditBureauToken = this.thitsaWorksCreditBureauIntegrationWritePlatformService
+                // .createToken(creditBureauID);
+
+                CreditBureauReportData reportobj = this.thitsaWorksCreditBureauIntegrationWritePlatformService
+                        .getCreditReportFromThitsaWorks(command);
+
+                // return new
+                // CommandProcessingResultBuilder().withCreditReport(reportobj).withCreditBureauToken(creditBureauToken).build();
+                return new CommandProcessingResultBuilder().withCreditReport(reportobj).build();
+            }
+
+            baseDataValidator.reset().failWithCode("creditBureau.has.not.been.Integrated");
+            throw new PlatformApiDataValidationException("creditBureau.has.not.been.Integrated", "creditBureau.has.not.been.Integrated",
+                    dataValidationErrors);
+
+        } catch (final DataIntegrityViolationException dve) {
+            handleTokenDataIntegrityIssues(command, dve.getMostSpecificCause(), dve);
+            return CommandProcessingResult.empty();
+        } catch (final PersistenceException ee) {
+            Throwable throwable = ExceptionUtils.getRootCause(ee.getCause());
+            handleTokenDataIntegrityIssues(command, throwable, ee);
+            return CommandProcessingResult.empty();
+        }
+
+    }
+
+    private Optional<String> getCreditBureau(Long creditBureauID) {
+
+        if (creditBureauID != null) {
+            Optional<CreditBureau> creditBureau = this.creditBureauRepository.findById(creditBureauID);
+
+            if (creditBureau.isEmpty()) {
+                return Optional.empty();
+            }
+
+            return Optional.of(creditBureau.get().getName());
+
+        }
+
+        return Optional.empty();
+    }
+
+    // saving the fetched creditreport in database
+    @Override
+    @Transactional
+    public CommandProcessingResult saveCreditReport(Long creditBureauId, String creditReportNumber, JsonCommand command) {
+
+        try {
+            this.context.authenticatedUser();
+
+            Optional<String> creditBureauName = getCreditBureau(creditBureauId);
+            CreditReport creditReport = null;
+
+            if (Objects.equals(creditBureauName.get(), CreditBureauConfigurations.THITSAWORKS.toString())) {
+
+                creditReport = creditReportRepository.getThitsaWorksCreditReport(creditBureauId, creditReportNumber);
+
+                // checks whether the creditReport for same NRC was saved before. if yes, then deletes it and replaces
+                // it with new one.
+                if (creditReport != null) {
+                    this.creditReportRepository.delete(creditReport);
+                    creditReport = null;
```

Comment:
don't set it as null and remove the null check below

Reply:
comment addressed.